base/compiler/tapirpasses.jl

```diff
@@ -0,0 +1,1033 @@
+    # syncregion is not used in nested tasks)
+
+    original_outputs = BitSet()
+    output_loader = zeros(Int, length(ir.stmts))
+    for i in 1:length(ir.stmts)
+        inst = ir.stmts[i][:inst]
+        if inst isa SyncNode
+            syncregion = (inst.syncregion::SSAValue).id
+            tg = get(taskgroups, syncregion, nothing)
+            if tg !== nothing
+                insert_node!(ir, i, Any, Expr(:call, Tapir.sync!, tg::SSAValue))
+                next_bb = block_for_inst(ir.cfg, i + 1)
+                ir.stmts.inst[i] = GotoNode(next_bb)
+                # i.e., transform
+                #     %syncregion = Expr(:syncregion)
+                #     ...
+                #     sync within %syncregion
+                # to
+                #     %tg = Tapir.taskgroup()
+                #     ...
+                #     Tapir.sync!(%tg)
+                #     goto #next_bb
+                #
+                # Note: using the "noop" goto node to avoid changing the CFG
+
+                tbl = get(tobeloaded, syncregion, nothing)
+                if tbl !== nothing
+                    bb = ir.cfg.blocks[block_for_inst(ir.cfg, i)]
+                    for (iout, T, ref) in tbl
+                        ex = Expr(:call, getfield, ref, QuoteNode(:x))
+                        load = insert_node!(ir, i, T, ex)
+                        @assert load.id > 0
+                        output_loader[iout] = load.id
+                        push!(original_outputs, iout)
+                    end
+                end
+            end
+        end
+    end
+
+    for i in 1:length(ir.stmts)
+        stmt = ir.stmts[i]
+        replaced = RefValue(false)
+        user = map_id(identity, stmt[:inst]) do v
+            if v isa SSAValue
+                if v.id in original_outputs
+                    replaced.x = true
+                    return SSAValue(output_loader[v.id])
+                end
+            end
+            return v
+        end
+        if replaced.x
+            stmt[:inst] = user
+        end
+    end
+
+    for i in ir.cfg.blocks[1].stmts
+        stmt = ir.stmts[i]
+        inst = stmt[:inst]
+        isexpr(inst, :invoke) && inst.args[2] === Tapir.noinline_box || continue
+        stmt[:inst] = Expr(:call, Tapir.box, inst.args[3])
+    end
+
+    # Finalize the changes in the IR (clears the node inserted to `ir.new_nodes`):
+    ir = compact!(ir, true)
+    if JLOptions().debug_level == 2
+        verify_ir(ir)
+        verify_linetable(ir.linetable)
+    end
+    # Remove the dead code in the detached sub-CFG (child tasks):
+    ir = cfg_simplify!(ir)
+    return ir
+end
+
+function code_info_from_ssair(ir::IRCode)
+    if JLOptions().debug_level == 2
+        verify_ir(ir)
+        verify_linetable(ir.linetable)
+    end
+    nargs = length(ir.argtypes)
+    ci = ccall(:jl_new_code_info_uninit, Ref{CodeInfo}, ())
+    # ci.slotnames = [Symbol(:arg, i - 1) for i in 1:nargs]  # how do I concatenate strings in Julia?
```

Comment:
Does `Symbol(:a, 1)` not work in the Core world?

Reply:
It looks like not.